src/OrbitQt/opengldetect.cpp

```diff
@@ -37,7 +67,21 @@ std::optional<OpenGlVersion> DetectOpenGlVersion() {
     return std::nullopt;
   }
 
-  return OpenGlVersion{gl_context.format().majorVersion(), gl_context.format().minorVersion()};
-}
+  std::string gl_version = reinterpret_cast<const char*>(glGetString(GL_VERSION));
+  std::vector<std::string> version_tokenized = absl::StrSplit(gl_version, '.');
+
+  if (version_tokenized.size() < 2) {
+    return std::nullopt;
+  }
+
+  int major_version;
+  absl::SimpleAtoi(version_tokenized[0], &major_version);
+  int minor_version;
+  absl::SimpleAtoi(version_tokenized[1], &minor_version);
```

Comment:
The `SimpleAtoi` calls could fail - theoretically. Probably they won't, but while reading that code the question why comes up. I suggest either an explaining comment or handling the failed-case.

Personally I would put these 10 lines into its own function that returns a `std::optional<OpenGlVersion>`. And I would return `std::nullopt` if either of the `SimpleAtoi` calls fails (returns false).

Reply:
It seems that clang agrees with me. `SimpleAtoi` is apparently marked as nodiscard.

Reply:
Yes, you are right of course! Fixed.